javascript/javascript3/week1/weatherApp.js

```diff
@@ -0,0 +1,156 @@
+      co_ordinates.style.visibility = "hidden";
+    h2.textContent = `Entered '${userInput.value}' ${data.message}`;
+
+  }
+  else
+  {
+    //weatherinfo will visible without loading the page after hidden
+    city.style.visibility = "visible";
+    temperature.style.visibility = "visible";
+    weatherIcon.style.visibility = "visible";
+      wind.style.visibility = "visible";
+      cloudy.style.visibility = "visible";
+      sunrise.style.visibility = "visible";
+      sunset.style.visibility = "visible";
+      co_ordinates.style.visibility = "visible";
+  city.textContent= data.name;
+  const actualTemp  = data.main.temp-273.15;
+//  console.log(actuaTemp);
+  temperature.textContent= `Temperature: ${actualTemp.toFixed(0)}°C`;
+  cloudy.textContent= `${data.weather[0].description}`;
+  wind.textContent=`Wind speed:${data.wind.speed}m/s`;
+  weatherIcon.src=`http://openweathermap.org/img/wn/${data.weather[0].icon}.png`;
+ weatherIcon.style.height='100px';
+  const sunriseInSeconds=(data.sys.sunrise*1000);
+  const sunsetInSeconds=(data.sys.sunset*1000);
+  const sunriseDate=new Date(sunriseInSeconds);
+  const sunsetDate=new Date(sunsetInSeconds);
+  sunrise.textContent=`SunRise:${sunriseDate.toLocaleTimeString()}`;
+  sunset.textContent=`SunSet:${sunsetDate.toLocaleTimeString()}`;
+  //for map
+  co_ordinates.textContent= 'Latitude: '+ data.coord.lat +' , Longitude: ' + data.coord.lon;
+}
+}).catch(err=>{
+    console.log('srr' + err);
+})
+}
+
+
+function getCurrentWeatherByCityName(){
+  h2.textContent = '';
+  const cityName = userInput.value;
+  localStorage.setItem("storedCityName", cityName);
+//  console.log('storedCityName: ' + storedCityName);
+  if(cityName === ''){
+    h2.textContent='Please enter city Name';
+  }else{
+weatherUrl =`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=a475908923b60eeeafd3fd4ce054d90e`;
+getWeather();
+}
+}
+
+function currentPositionWeatherByLocation() {
+  let latitude;
+ let longitude;
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function showPosition(position) {
+      const lat = position.coords.latitude;
+      const lon= position.coords.longitude;
+      co_ordinates.innerHTML = "Latitude: " + position.coords.latitude +
+      "<br>Longitude: " + position.coords.longitude;
+      map = new google.maps.Map(document.getElementById("map"), {
+        center: { lat: position.coords.latitude, lng: position.coords.longitude},
+        zoom: 8,
+      /*const map = new google.maps.Map(document.getElementById("map")), {
+            center: { lat: position.coords.latitude, lon: position.coords.longitude   },
+            zoom: 8,*/
+        });
+       weatherUrl = `http://api.openweathermap.org/data/2.5/find?lat=${lat}&lon=${lon}&cnt=1&appid=a475908923b60eeeafd3fd4ce054d90e`;
+
+        fetch(weatherUrl)
+        .then(response => response.json())
+        .then(data=>{
+          console.log(data);
+          city.textContent= data.list[0].name;
```

Comment:
Here you are rewriting the code so better to write all the code in one function the whenever you want so show that data you just call the function .